File: 09_dynamic_programming/golang/01_longest_common_subsequence.go
package main
import "fmt"

func main(){
	if word_a[i] == word_b[j] {
		cell[i][j] = cell[i-1][j-1]+1
	}else{
		cell[i][j] = max(cell[i-1][j], cell[i][j-1])
	}
}

File: 09_dynamic_programming/elixir/01_longest_common_subsequence.exs
# `cell` is a map of maps here, e.g.
# cell = %{
#   0 => %{0 => 0, ...},
#   1 => %{...},
#   ...
# }

if String.at(word_a, i) == String.at(word_a, j) do
  put_in(cell[i - 1][j - 1], cell[i - 1][j - 1] + 1)
else
  put_in(cell[i - 1][j - 1], max(cell[i - 1][j], cell[i][j - 1]))
end
